hart_protocol/_parsing.py

```diff
@@ -11,14 +11,18 @@ def parse(response: bytes) -> MutableMapping[str, Union[int, bytes, str, float]]
     else:  # short address
         out["address"] = response[1]
         response = response[2:]
-    command, bytecount, status = struct.unpack_from(">BBL", response)
+    command, bytecount, status = struct.unpack_from(">BBH", response)
     out["status"] = status
```

Comment:
```suggestion
    command, bytecount, status, device_status = struct.unpack_from(">BBHH", response)
    out["status"] = status
    out["device_status"] = status
```

From my reading of the protocol, there are always two bytes of status information in the response. First byte is communication status, second is device status.

I think you're right that we should change this behavior---it's confusing to pack those two unrelated status bytes into a single long. However, I'd like to somehow continue to expose the device_status.

Reply:
Yes, that's right, and even better. But then we have to unpack the two bytes into bytes, like:

```suggestion
    command, bytecount, response_code, status = struct.unpack_from(">BBBB", response)
    out["status"] = status
```

Reply:
I guess response code is a beter name to use for the first byte, but I'll look this up in a minute

Reply:
The first status byte (I've proposed `response_code`) is in fact a union
```
    union {
        uint8_t communication_status;
        uint8_t command_response_code;
    };
```

In case there are communication errors then the communication status should be returned, instead of the command response code. See https://library.fieldcommgroup.org/20085/TS20085/3.0.1/#page=44.